dask/dataframe/groupby.py

```diff
@@ -1124,6 +1124,14 @@ def _groupby_kwargs(self):
             **self.observed,
         }
 
+    def __iter__(self):
+        raise NotImplementedError(
+            "Iteration of DataFrameGroupBy objects requires computing the groups which "
+            "may be slow. You probably want to use 'apply' to execute a function for "
+            "all the columns. To access individual groups, use 'get_group'. To list "
+            "all the group names, use 'df[<group column>].unique().compute()'."
```

Comment:
```suggestion
            "may be slow. You may want to use 'apply' or 'transform' to execute a function for "
            "all the groups. To access individual groups, use 'get_group'. To list "
            "all the group names, use 'df.groupby(<group-columns-or-index>).size().compute()'."
```

Suggesting a different way to get the group names which should also work for multiple columns or the index.

Reply:
Thanks @ian-r-rose! The only quibble I have is with the last change from `.unique()` to `.size()`. I think the former would give you the values of the groups to put into `get_group()' whereas the latter just tells you how many rows are in the dataset right?